cpu/esp32/ld/esp32c3/sections.ld

```diff
@@ -223,7 +223,7 @@ SECTIONS
     *components/spi_flash/spi_flash_chip_winbond.*(.literal .literal.* .text .text.*)
     *components/spi_flash/*/spi_flash_rom_patch.*(.literal .literal.* .text .text.*)
     *components/spi_flash/spi_flash_timing_tuning.*(.literal .literal.* .text .text.*)
-    *components/spi_flash/spi_timing_config.*(.literal .literal.* .text .text.*)
+    *components/spi_flash/*/spi_timing_config.*(.literal .literal.* .text .text.*)
```

Comment:
What's that change about?

Reply:
`spi_timing_config.c` is not used by ESP32 and ESP32-C3. Therefore I did not realize the wrong path before. As I mentioned before, the intention is to use the same list of objects in all linker scripts to ease maintenance, even if not all listed objects are used by all ESP32x SoC variants.